commit 14e1391b71027948cdbacdbea4bf8858c2068eb7
Author: Arend van Spriel <arend.vanspriel@broadcom.com>
Date:   Wed Jan 3 10:57:02 2024 +0100

    wifi: brcmfmac: add per-vendor feature detection callback
    
    Adding a .feat_attach() callback allowing per-vendor overrides
    of the driver feature flags. In this patch the callback is only
    provided by BCA vendor to disable SAE feature as it has not been
    confirmed yet. BCA chips generally do not have the in-driver
    supplicant (idsup) feature so they rely on NL80211_CMD_EXTERNAL_AUTH
    to trigger user-space authentication.
    
    Signed-off-by: Arend van Spriel <arend.vanspriel@broadcom.com>
    Signed-off-by: Kalle Valo <kvalo@kernel.org>
    Link: https://msgid.link/20240103095704.135651-3-arend.vanspriel@broadcom.com

diff --git a/drivers/net/wireless/broadcom/brcm80211/brcmfmac/bca/core.c b/drivers/net/wireless/broadcom/brcm80211/brcmfmac/bca/core.c
index ac3a36fa3640..a5d9ac5e6763 100644
--- a/drivers/net/wireless/broadcom/brcm80211/brcmfmac/bca/core.c
+++ b/drivers/net/wireless/broadcom/brcm80211/brcmfmac/bca/core.c
@@ -7,6 +7,7 @@
 #include <core.h>
 #include <bus.h>
 #include <fwvid.h>
+#include <feature.h>
 
 #include "vops.h"
 
@@ -21,7 +22,14 @@ static void brcmf_bca_detach(struct brcmf_pub *drvr)
 	pr_err("%s: executing\n", __func__);
 }
 
+static void brcmf_bca_feat_attach(struct brcmf_if *ifp)
+{
+	/* SAE support not confirmed so disabling for now */
+	ifp->drvr->feat_flags &= ~BIT(BRCMF_FEAT_SAE);
+}
+
 const struct brcmf_fwvid_ops brcmf_bca_ops = {
 	.attach = brcmf_bca_attach,
 	.detach = brcmf_bca_detach,
+	.feat_attach = brcmf_bca_feat_attach,
 };